plinth/modules/matrixsynapse/views.py

```diff
@@ -74,3 +79,27 @@ def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(**kwargs)
         context['domain_name'] = matrixsynapse.get_configured_domain_name()
         return context
+
+    def get_initial(self):
+        """Return the values to fill in the form."""
+        initial = super().get_initial()
+        initial.update({
+            'enable_public_registration': get_public_registration_status()})
+        return initial
+
+    def form_valid(self, form):
+        """Handle valid form submission."""
+        old_config = self.get_initial()
+        new_config = form.cleaned_data
+        if old_config['enable_public_registration'] != new_config['enable_public_registration']:
+            enable_registration = new_config['enable_public_registration']
+            if enable_registration:
+                actions.superuser_run('matrixsynapse', ['enable-registration'])
+                messages.success(self.request,
+                                 _('Public registration enabled'))
+            else:
+                actions.superuser_run('matrixsynapse', ['disable-registration'])
+                messages.success(self.request,
+                                 _('Public registration disabled'))
+
+        return super().form_valid(form)
```

Comment:
Another consideration/question: Should synapse be restarted in case of a config change, such that it takes effect? Or maybe another kill signal (might be faster)? Or maybe put this into the action itself, in case an admin calls it from the command line?

Reply:
I don't think the server does not supports automatically rereading configuration file when it is changed.  It appears that matrix-synapse server also does not support reloading (sending a signal to ask for configuration reload):
```bash
/usr/bin/python -m synapse.app.homeserver --help
cat /lib/systemd/system/matrix-synapse.service
```
So restart is the way to go.